File: picasso/src/test/java/com/squareup/picasso3/AssetRequestHandlerTest.java
package com.squareup.picasso3;

import android.content.Context;
import android.net.Uri;
import java.io.IOException;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.Mock;
import org.robolectric.RobolectricTestRunner;

import static com.google.common.truth.Truth.assertThat;
import static org.mockito.MockitoAnnotations.initMocks;

@RunWith(RobolectricTestRunner.class)
public class AssetRequestHandlerTest {
  @Mock Context context;

  @Before public void setUp() {
    initMocks(this);
  }

  @Test public void truncatesFilePrefix() throws IOException {
    Uri uri = Uri.parse("file:///android_asset/foo/bar.png");
    Request request = new Request.Builder(uri).build();

    String actual = AssetRequestHandler.getFilePath(request);
    assertThat(actual).isEqualTo("foo/bar.png");
  }
}

File: picasso/src/test/java/com/squareup/picasso3/AssetRequestHandlerTest.kt
package com.squareup.picasso3

import android.net.Uri
import com.google.common.truth.Truth.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import java.io.IOException

@RunWith(RobolectricTestRunner::class)
class AssetRequestHandlerTest {

  @Test
  @Throws(IOException::class)
  fun truncatesFilePrefix() {
    val uri = Uri.parse("file:///android_asset/foo/bar.png")
    val request = Request.Builder(uri).build()

    val actual = AssetRequestHandler.getFilePath(request)
    assertThat(actual).isEqualTo("foo/bar.png")
  }
}